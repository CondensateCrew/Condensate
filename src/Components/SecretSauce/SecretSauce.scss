@import '../../css/index.scss';
@import '../../css/variables.scss';

.secret-sauce-section {
  @extend %one-line-block;
  padding: $one-line-padding-space;
}

.secret-sauce-header-div {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  h3 {
    font-weight: 400;
    width: 100%;
  }
  button {
    display: -webkit-flex;
    display: -ms-flex;
    display: flex;
    align-items: center;
    justify-content: center;
    padding: $extra-small-space;
    background: $primary-gradient;
    border-radius: $one-line-border-radius;
    box-shadow: $external-shadow;
  }
  img {
    width: $big-text;
    &.up-btn {
      transform: rotate(180deg);
    }
  }
}

.secret-sauce-display-div {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  margin: $small-space 0;
  h2 {
    width: 80%;
    box-shadow: $inner-shadow;
    padding: $small-space;
    border-radius: $one-line-border-radius;
  }
  img {
    width: $big-text;
    margin-left: $small-space;
    opacity: 0.4;
    @extend %hover-transform;
  }
}
