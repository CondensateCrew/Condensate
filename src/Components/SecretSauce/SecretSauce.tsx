import React, { useState } from 'react';
import './SecretSauce.scss';
import down from '../../assets/down.svg';
import refresh from '../../assets/refresh.svg';
import { useSelector } from 'react-redux';
import { AppStore } from 'interfaces';

const SecretSauce:React.FC = () => {
  const [ currentStep, setCurrentStep ] = useState<number>(0);
  const [ isClicked, setIsClicked ] = useState<boolean>(false);

  const secretSauce  = useSelector((state:AppStore) => state.secretSauce);
  const refreshSecretSauce = ():void => {
    if (currentStep < 20) {
      setCurrentStep(currentStep + 1);
    } else {
      setCurrentStep(0);
    }
  }

  const toggleBlock = () => setIsClicked(!isClicked);
  const upClass = (isClicked) ? 'up-btn' : '';

  return (
    <section className='secret-sauce-section'>
      <header className='secret-sauce-header-div'>
        <h3>Random Word</h3>
        <button onClick={toggleBlock}>
          <img src={down} alt='down-arrow' className={upClass} />
        </button>
      </header>
      { isClicked &&
        <main className='secret-sauce-display-div'>
          <h2>{secretSauce[currentStep]}</h2>
          <img onClick={refreshSecretSauce} src={refresh} alt='refresh-recycle icon'/>
        </main>
      }
    </section>
  )
}

export default SecretSauce;
