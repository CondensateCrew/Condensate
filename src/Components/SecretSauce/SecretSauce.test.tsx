import React from 'react';
import SecretSauce from './SecretSauce';
import { shallow } from 'enzyme';

const mockDispatch = jest.fn()
jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
  useSelector: () => ['Goose']
}));

describe('SecretSauce component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<SecretSauce />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if block is opened', () => {
    const wrapper = shallow(<SecretSauce />);
    wrapper.find('header>button').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
