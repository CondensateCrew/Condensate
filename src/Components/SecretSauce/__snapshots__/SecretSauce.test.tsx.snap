// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecretSauce component should match the snapshot 1`] = `
<section
  className="secret-sauce-section"
>
  <header
    className="secret-sauce-header-div"
  >
    <h3>
      Random Word
    </h3>
    <button
      onClick={[Function]}
    >
      <img
        alt="down-arrow"
        className=""
        src="down.svg"
      />
    </button>
  </header>
</section>
`;

exports[`SecretSauce component should match the snapshot if block is opened 1`] = `
<section
  className="secret-sauce-section"
>
  <header
    className="secret-sauce-header-div"
  >
    <h3>
      Random Word
    </h3>
    <button
      onClick={[Function]}
    >
      <img
        alt="down-arrow"
        className="up-btn"
        src="down.svg"
      />
    </button>
  </header>
  <main
    className="secret-sauce-display-div"
  >
    <h2>
      Goose
    </h2>
    <img
      alt="refresh-recycle icon"
      onClick={[Function]}
      src="refresh.svg"
    />
  </main>
</section>
`;
